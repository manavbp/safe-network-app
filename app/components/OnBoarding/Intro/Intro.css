.Base {
    width: 100%;
    height: 100%;
    background-image: url('../../../assets/images/on_boarding_popup_bg.png');
    background-repeat: no-repeat;
    background-size: contain;
    display: 'flex';
    justify-content: center;
    align-items: center;
    flex-direction: column;
}

.Box {
    padding: '0 24px';
}

.Title {
    padding-top: 450px;
    text-align: center;
    margin-bottom: 16px;
}

.Description {
    text-align: center;
}
