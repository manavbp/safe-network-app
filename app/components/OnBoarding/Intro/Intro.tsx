import React from 'react';
import Typography from '@material-ui/core/Typography';
import Paper from '@material-ui/core/Paper';
import Box from '@material-ui/core/Box';
import styles from './Intro.css';

export const Intro = () => {
    return (
        <Paper className={styles.Base} elevation={0} aria-label="IntroPage">
            <Box className={styles.Container}>
                <Typography className={styles.Title} variant="h5">
                    One Place for All SAFE Apps
                </Typography>
                <Typography className={styles.Description}>
                    A one-stop shop to access all SAFE Apps and manage instant
                    app updates.
                </Typography>
            </Box>
        </Paper>
    );
};
