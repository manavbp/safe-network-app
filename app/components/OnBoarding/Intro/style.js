import { styled } from '@material-ui/core/styles';
import Paper from '@material-ui/core/Paper';
import Box from '@material-ui/core/Box';
import Typography from '@material-ui/core/Typography';

export const Base = styled( Paper )( {
    width: '100%',
    height: '675%',
    backgroundImage: `url(${'assets/images/on_boarding_popup_bg.png'})`,
    backgroundRepeat: 'no-repeat',
    backgroundSize: 'contain',
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
    flexDirection: 'column'
} );

export const Container = styled( Box )( {
    padding: '0 24px',
    textAlign: 'center'
} );

export const Title = styled( Typography )( {
    paddingTop: '450px',
    marginBottom: '16px'
} );
