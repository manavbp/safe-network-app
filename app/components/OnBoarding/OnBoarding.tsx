import React, { Component } from 'react';
import { History } from 'history';
import Grid from '@material-ui/core/Grid';
import { Switch, Route } from 'react-router';
import { styled } from '@material-ui/core/styles';
import {
    UserPreferences,
    AppPreferences,
    Preferences,
    LaunchpadState
} from '$Definitions/application.d';

import { Stepper } from './Stepper/Stepper';
import { GetStarted } from './GetStarted';
import { Intro } from './Intro';
import { BasicSettings } from './BasicSettings/BasicSettings';
import {
    HOME,
    INTRO,
    BASIC_SETTINGS,
    ON_BOARDING
} from '$Constants/routes.json';

interface Props {
    setAppPreferences: Function;
    aliasShouldOnboard: Function;
    triggerSetAsTrayWindow: Function;
    userPreferences: UserPreferences;
    appPreferences: AppPreferences;
    setUserPreferences: Function;
    getUserPreferences: Function;
    storePreferences: Function;
    isTrayWindow: boolean;
    setOnboardCompleted: Function;
    autoLaunch: Function;
    history?: History;
}

const Base = styled( Grid )( {
    position: 'absolute',
    top: 0,
    left: 0,
    width: '100%',
    height: '100%'
} );

export class OnBoarding extends React.Component<Props> {
    state = {
        currentPosition: 0
    };

    totalSteps = 3;

    componentWillMount() {
        const { appPreferences, history, getUserPreferences } = this.props;

        if ( !appPreferences.shouldOnboard ) {
            history.push( HOME );
        }
        getUserPreferences();
    }

    completed = () => {
        const {
            storePreferences,
            userPreferences,
            autoLaunch,
            setAppPreferences
        } = this.props;

        const appPreferences: AppPreferences = {
            shouldOnboard: false
        };

        const newUserPreferences: UserPreferences = {
            ...userPreferences
        };

        const preferences: Preferences = {
            userPreferences: { ...newUserPreferences },
            appPreferences
        };

        storePreferences( preferences );
        setAppPreferences( { shouldOnboard: false } );

        if ( userPreferences.launchOnStart ) {
            autoLaunch( true );
        }
    };

    onNext = () => {
        const { currentPosition } = this.state;
        const position = currentPosition;

        if ( currentPosition < this.totalSteps - 1 ) {
            this.setState( {
                currentPosition: currentPosition + 1
            } );
        } else {
            this.completed();
        }
    };

    onBack = () => {
        const { currentPosition } = this.state;
        const position = currentPosition;

        if ( currentPosition > 0 ) {
            this.setState( {
                currentPosition: currentPosition - 1
            } );
        }
    };

    render() {
        const { currentPosition } = this.state;
        const {
            userPreferences,
            setUserPreferences,
            getUserPreferences,
            storePreferences,
            isTrayWindow,
            triggerSetAsTrayWindow,
            autoLaunch,
            history
        } = this.props;

        const isGetStarted = currentPosition === 0;

        return (
            <Base container>
                <Grid item xs={12}>
                    <Switch>
                        <Route
                            exact
                            path={ON_BOARDING}
                            component={() => (
                                <GetStarted
                                    onClickGetStarted={this.onNext}
                                    // @ts-ignore
                                    history={history}
                                />
                            )}
                        />
                        <Route path={INTRO} component={() => <Intro />} />
                        <Route
                            path={BASIC_SETTINGS}
                            component={() => (
                                <BasicSettings
                                    userPreferences={userPreferences}
                                    setUserPreferences={setUserPreferences}
                                    triggerSetAsTrayWindow={
                                        triggerSetAsTrayWindow
                                    }
                                    autoLaunch={autoLaunch}
                                    isTrayWindow={isTrayWindow}
                                />
                            )}
                        />
                    </Switch>
                </Grid>
                <Stepper
                    theme={isGetStarted ? 'white' : 'default'}
                    showButtons={!isGetStarted}
                    noElevation={isGetStarted}
                    onNext={this.onNext}
                    onBack={this.onBack}
                    steps={this.totalSteps}
                    activeStep={this.state.currentPosition}
                    // @ts-ignore
                    history={history}
                />
            </Base>
        );
    }
}
