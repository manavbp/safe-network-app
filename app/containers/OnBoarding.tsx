import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import { OnBoarding } from '$Components/OnBoarding/OnBoarding';

import {
    setUserPreferences,
    setAppPreferences,
    getUserPreferences
} from '$Actions/launchpad_actions';
import {
    autoLaunch,
    triggerSetAsTrayWindow
} from '$Actions/alias/launchpad_actions';

const mapStateToProperties = ( state ) => {
    return {
        userPreferences: state.launchpad.userPreferences,
        appPreferences: state.launchpad.appPreferences,
        isTrayWindow: state.launchpad.isTrayWindow
    };
};

const mapDispatchToProperties = ( dispatch ) => {
    const actions = {
        setAppPreferences,
        setUserPreferences,
        getUserPreferences,
        autoLaunch,
        triggerSetAsTrayWindow
    };
    return bindActionCreators( actions, dispatch );
};

export const OnBoardingPage: React.ComponentClass = connect(
    mapStateToProperties,
    mapDispatchToProperties
)( OnBoarding );
