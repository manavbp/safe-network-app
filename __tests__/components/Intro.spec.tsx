import React from 'react';
import { createShallow } from '@material-ui/core/test-utils';

import { Intro } from '$App/components/OnBoarding/Intro';

const shallow = createShallow();

const shallowSetup = ( propOverrides? ) => {
    const props = Object.assign( {}, propOverrides );

    const wrapper = shallow( <Intro {...props} /> );

    return {
        props,
        wrapper
    };
};

describe( 'Intro', () => {
    it( 'render', () => {
        const { wrapper } = shallowSetup();
        expect( wrapper ).toMatchSnapshot();
    } );

    it( 'title should be right', () => {
        const { wrapper } = shallowSetup();
        expect(
            wrapper
                .childAt( 0 )
                .childAt( 0 )
                .text()
        ).toEqual( 'One Place for All SAFE Apps' );
        expect(
            wrapper
                .childAt( 0 )
                .childAt( 0 )
                .prop( 'variant' )
        ).toEqual( 'h5' );
    } );

    it( 'description should be right', () => {
        const { wrapper } = shallowSetup();
        expect(
            wrapper
                .childAt( 0 )
                .childAt( 1 )
                .text()
        ).toEqual(
            'A one-stop shop to access all SAFE Apps and manage instant app updates.'
        );
    } );
} );
