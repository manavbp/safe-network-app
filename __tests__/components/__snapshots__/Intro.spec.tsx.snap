// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Intro render 1`] = `ShallowWrapper {}`;
